src/loginAzurePowerShell.ts

```diff
@@ -0,0 +1,76 @@
+import * as os from 'os';
+import * as core from '@actions/core';
+import * as exec from '@actions/exec';
+import * as io from '@actions/io';
+
+var psPath: string;
+
+export const initializeAz = async (servicePrincipalId: string, servicePrincipalKey: string, tenantId: string, subscriptionId: string) => {
```

Comment:
I think a better approach would be to define an interface IAzurePowerShellSession . It should have 2 methods: Initialize and Login. Implement a class. we can discuss on this.